src/avx2/stage1.rs

```diff
@@ -438,7 +474,7 @@ impl<'de> Deserializer<'de> {
                 .copy_from(input.as_ptr().add(idx), len as usize - idx);
             let input: SimdInput = fill_input(&tmpbuf);
 
-            check_utf8(&input, &mut has_error, &mut previous);
+            check_utf8(&input, &mut utf8_state);
```

Comment:
We should consider returning the register instead of passing it, Lemire had a nice [blog article](https://lemire.me/blog/2019/09/05/passing-integers-by-reference-can-be-expensive/) on passing integers by ref vs returning them, it might apply for simd registers as well.

Reply:
I like this idea. I'm also torn because I'd love to have the core engine correspond more closely to simdjson releases. I wonder if we can provide an easy way to include this type of implementation refinement after this PR lands and the template/generic PR refactoring is done.

Reply:
@lemirehas always been open to taking contributions back I think if it's sensible we can just push it back to upstream :) win for all!